gwcs/wcs.py

```diff
@@ -634,3 +641,235 @@ def fix_inputs(self, fixed):
         new_pipeline.append((step0[0], new_transform))
         new_pipeline.extend(self.pipeline[1:])
         return self.__class__(new_pipeline)
+
+    def to_fits_sip(self, bounding_box, max_error=None, max_rms=None, degree=None,
+                    max_inv_error=None, max_inv_rms=None, verbose=False):
+        """
+        Make a SIP-based approximation to the WCS to be used in FITS WCS
+
+        Parameters
+        ----------
+        bounding_box: a pair of tuples, each consisting of two integers
+            Represents the range of pixel values in both dimensions
+            ((ymin, ymax), (xmin, xmax))
+        max_error: float
+            Maximum allowed error over the domain of the pixel array.
+        max_rms_error: float
+            Maximum allowed rms error over the domain of the pixel array.
+        degree: integer
+            Degree of the SIP polynomial. If supplied, max_error and max_rms_error
+            must be None.
+        max_inv_error: float
+            Maximum allowed inverse error over the domain of the pixel array.
+        max_inv_rms: float
+            Maximum allowed inverse rms over the domain of the pixel array
+        verbose: boolean
+            print progress of fits
+
+        Returns
+        -------
+        fits_dict, max_error, rms_error
+
+        fits_dict is a dictionary where the keys are FITS keywords and the values
+        the corresonding FITS keyword values, to make it easy to merge with
+        an existing header.
+
+        This assumes a tangent projection.
+        """
+        if (self.forward_transform.n_inputs !=2
+            or self.forward_transform.n_outputs != 2):
+           raise ValueError("The to_fits_sip method only works with 2 dimensional transforms")
+
+        # Handle the options related to degree and thresholds.
+        if degree is not None and (max_error is not None or max_rms is not None):
+            raise ValueError(
+                "Degree cannot be specified if either max_error or max_rms have values")
+        if degree is None and max_error is None and max_rms is None:
+            raise ValueError(
+                "At least one of max_error, max_rms, or degree must be specified")
+        fits_dict = {}
+        transform = self.forward_transform
+        # Determine reference points.
+        (ymin, ymax), (xmin, xmax) = bounding_box
+        crpix1 = int((xmax - xmin) / 2)
```

Comment:
I do not think `CRPIX` is _required_ to be `int`. Actually, this approach (with trying to determine `CRPIX` from `bounding_box`) is a little worrisome because `SIP`'s `CRPIX` and "parent" `WCS`' `CRPIX` **must** be the same. So, I feel this function **needs** `CRPIX` as input instead of trying to guess it.

Reply:
As I understand it, the parent WCS (GWCS actually) isn't required to have a CRPIX. A smarter version could try to examine the original WCS to see if there is an equivalent, but that can be fairly involved if the first transforms don't fit the expected form.

Reply:
Well, GWCS is not required to have one but the FITS WCS that you are constructing _is required_ to have one => SIP's CRPIX must match the CRPIX of the actual parent WCS (FITS WCS in this case).

Reply:
Ah, I should have seen that you are actually constructing the _entire_ FITS WCS here. => disregard my previous comments about CRPIX.